src/java/org/apache/cassandra/net/CustomParamsSerializer.java

```diff
@@ -0,0 +1,80 @@
+
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.net;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Maps;
+
+import org.apache.cassandra.db.TypeSizes;
+import org.apache.cassandra.io.IVersionedSerializer;
+import org.apache.cassandra.io.util.DataInputPlus;
+import org.apache.cassandra.io.util.DataOutputPlus;
+import org.apache.cassandra.utils.ByteBufferUtil;
+
+
+class CustomParamsSerializer implements IVersionedSerializer<Map<String,byte[]>>
+{
+    public static final CustomParamsSerializer serializer = new CustomParamsSerializer();
+
+    @Override
+    public void serialize(Map<String, byte[]> t, DataOutputPlus out, int version) throws IOException
+    {
+        out.writeInt(t.size());
+        for (Map.Entry<String, byte[]> e : t.entrySet())
+        {
+            out.writeUTF(e.getKey());
+            out.writeInt(e.getValue().length);
```

Comment:
I would use a varint here, as four bytes per value is going to be mostly wasteful in this context.

Reply:
Would throw the helper methods `writeWithVIntLength()`, `readWithVIntLength()`, and `sizeWithVIntLength()` into `ByteBufferUtil`, next to the existing ones for short/int. We have the equivalents for byte buffers in `ByteBufferUtil`, might as well add them for byte arrays.